controller/transfer/labelstudio/project_creation_manager.py

```diff
@@ -0,0 +1,208 @@
+import json
+from typing import Dict, Any, List, Tuple
+
+from controller.transfer.labelstudio.util import create_unknown_users
+from controller.transfer.record_transfer_manager import download_file
+from submodules.model import enums, events
+from submodules.model.business_objects import (
+    upload_task,
+    attribute,
+    labeling_task,
+    record,
+    record_label_association,
+    labeling_task_label,
+    general,
+    project,
+)
+from controller.upload_task import manager as upload_task_manager
+from util import doc_ock
+from controller.user import manager as user_manager
+
+
+def manage_data_import(project_id: str, task_id: str) -> None:
+    task = upload_task.get(project_id, task_id)
+    file_path = download_file(project_id, task)
+    mappings = json.loads(task.mappings)
+    user_mapping = mappings.get("users")
+    user_mapping = create_unknown_users(user_mapping)
+    attribute_task_mapping = mappings.get("tasks")
+    with open(file_path) as file:
+        data = json.load(file)
+
+        first_record_item = data[0]
+        for attribute_name, attribute_value in first_record_item.get("data").items():
+            __create_attribute(project_id, attribute_name, attribute_value)
+
+        labeling_tasks, records, record_label_associations = __extract_data(
+            data, user_mapping, attribute_task_mapping
+        )
+        label_id_lookup = __create_labeling_tasks(project_id, labeling_tasks)
+        __create_records(
+            project_id, records, record_label_associations, label_id_lookup
+        )
+        number_records = len(records)
+
+    upload_task_manager.update_upload_task_to_finished(task)
+    upload_task_manager.update_task(
+        project_id, task.id, state=enums.UploadStates.DONE.value, progress=100.0
+    )
+
+    user = user_manager.get_or_create_user(task.user_id)
+    project_item = project.get(project_id)
+    doc_ock.post_event(
+        user,
+        events.UploadRecords(
+            ProjectName=f"{project_item.name}-{project_item.id}", Records=number_records
+        ),
+    )
+    general.commit()
+
+
+def __create_records(
+    project_id: str,
+    records: List[Dict],
+    record_label_associations: List[Dict],
+    label_id_lookup: Dict,
+) -> None:
+    record_mapping_dict = {}
+
+    for record_item in records:
```

Comment:
i think we should at least chunk the commits a bit so e.g. every 500 records a commit

Reply:
did